src/file-upload/file-upload.service.ts

```diff
@@ -0,0 +1,140 @@
+import { Injectable } from "@nestjs/common";
+import { CreateFileUploadDto } from "./dto/create-file-upload.dto";
+import { PrismaService } from "src/prisma/prisma.service";
+import * as fs from "fs";
+import csvParser = require("csv-parser");
+
+@Injectable()
+export class FileUploadService {
+  constructor(private prisma: PrismaService) {}
+
+  public async storeData(data: CreateFileUploadDto[]) {
+    // Check if the competency Existed or not the given csv data if not then filter and throw error
+    const allCompetencies = await this.prisma.competency.findMany();
```

Comment:
Please use the existing code in their respective services for db queries here and the other places.

Reply:
modified the code by using the existing services